app/uk/gov/hmrc/cataloguefrontend/search/IndexBuilder.scala

```diff
@@ -0,0 +1,83 @@
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import uk.gov.hmrc.cataloguefrontend.healthindicators.{routes => healthRoutes}
+import uk.gov.hmrc.cataloguefrontend.leakdetection.{routes => leakRoutes}
+import uk.gov.hmrc.cataloguefrontend.teams.{routes => teamRoutes}
+import uk.gov.hmrc.cataloguefrontend.whatsrunningwhere.{routes => wrwRoutes}
+import uk.gov.hmrc.cataloguefrontend.{routes => catalogueRoutes}
+
+import java.util.concurrent.atomic.AtomicReference
+
+case class SearchTerm(name: String,  link: String, linkType: String) {
+  lazy val term: String = IndexBuilder.normalizeTerm(name)
+}
+
+@Singleton
+class IndexBuilder @Inject()(teamsAndRepositoriesConnector: TeamsAndRepositoriesConnector)(implicit ec: ExecutionContext){
+
+  val cachedIndex = new AtomicReference[Seq[SearchTerm]]()
+
+  val hardcodedLinks = List(
+    new SearchTerm("dependency explorer",      catalogueRoutes.DependencyExplorerController.landing.url, "explorer") {
+      override lazy val term: String = "depex"
+    },
+    SearchTerm("dependency explorer", catalogueRoutes.DependencyExplorerController.landing.url, "explorer"),
+    SearchTerm("bobby",      catalogueRoutes.BobbyExplorerController.list().url, "explorer"),
+    SearchTerm("jvm",        catalogueRoutes.JDKVersionController.compareAllEnvironments.url, "explorer"),
+    SearchTerm("leaks",      leakRoutes.LeakDetectionController.ruleSummaries.url, "explorer"),
+    SearchTerm("lds",        leakRoutes.LeakDetectionController.ruleSummaries.url, "explorer"),
+  )
+
+  def buildIndexes(): Future[List[SearchTerm]] = {
+    implicit val hc = HeaderCarrier()
+    for {
+      repos         <- teamsAndRepositoriesConnector.allRepositories
+      teams         <- teamsAndRepositoriesConnector.allTeams
+      teamPageLinks =  teams.flatMap(t => List(SearchTerm(t.name.asString, teamRoutes.TeamsController.team(t.name).url, "teams"),
+                                               SearchTerm(t.name.asString, url"${wrwRoutes.WhatsRunningWhereController.releases(false).url}?profile_type=team&profile_name=${t.name.asString}".toString, "deployments")))
+      repoLinks     =  repos.flatMap(r => List(SearchTerm(r.name, catalogueRoutes.CatalogueController.repository(r.name).url, "repo"),
+                                               SearchTerm(r.name, healthRoutes.HealthIndicatorsController.breakdownForRepo(r.name).url, "health"),
+                                               SearchTerm(r.name, leakRoutes.LeakDetectionController.report(r.name, r.defaultBranch).url,"leaks")))
+      serviceLinks  =  repos.filter(_.repoType == RepoType.Service).map(r => SearchTerm(r.name, catalogueRoutes.CatalogueController.serviceConfig(r.name).url, "config"))
+    } yield hardcodedLinks ++ teamPageLinks ++ repoLinks ++ serviceLinks
+  }
+
+  def getIndex(refresh: Boolean = false): Future[Seq[SearchTerm]] = {
```

Comment:
do we need the refresh flag? Thinking if we need a refresh, we just trigger the s`IndexScheduler` to run. That way we can guarantee we don't have it running twice at the same time etc.
Also since the scheduler runs after deployment (1 second) I don't think we need to worry about the cache being empty case either - if it's empty, it's more likely because the scheduler is already running?